client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -31,14 +31,11 @@ use crate::{
     models::devices::{Bios, Board, Chassis, Processor},
 };
 
-pub(crate) fn load_smbios_data(entry_filepath: &Path, table_filepath: &Path) -> Option<SMBiosData> {
-    match table_load_from_device(entry_filepath, table_filepath) {
-        Ok(data) => Some(data),
-        Err(e) => {
-            eprintln!("failed to load SMBIOS data: {}.", e);
-            None
-        }
-    }
+pub(crate) fn load_smbios_data(
+    entry_filepath: &Path,
+    table_filepath: &Path,
+) -> Result<SMBiosData, anyhow::Error> {
```

Comment:
I think the error type of this `Result` is equivalent to an `anyhow::Result`, so it's probably cleaner just to use that alias :)